src/transformer.ts

```diff
@@ -728,6 +729,79 @@ const injectItemsetTemplateCalls = (
     xslDoc.documentElement.append(...templateCalls);
 };
 
+/**
+ * This is a replacement for the logic previously implemented in the
+ * `appearance` template in `openrosa2html5form.xsl`. That template now
+ * transforms existing appearance values directly to a `data-appearance`
+ * attribute, and the previous logic is applied here based on those values.
+ */
+const transformAppearances = (doc: DOM.Document) => {
+    const appearanceElements = getNodesByXPathExpression(
+        doc,
+        '//*[@data-appearances]'
+    );
+
+    appearanceElements.forEach((element) => {
+        const selectType = element.hasAttribute('data-appearances-select-type');
+
+        if (selectType) {
+            element.removeAttribute('data-appearances-select-type');
+        }
+
+        const appearances =
+            element
+                .getAttribute('data-appearances')
+                ?.trim()
+                .toLowerCase()
+                .split(/\s+/) ?? [];
+
+        const appearanceClasses = appearances.flatMap((appearance) => {
+            const results = [`or-appearance-${appearance}`];
+
+            // Convert deprecated appearances, but leave the deprecated ones.
```

Comment:
This is a comment that existed before but I don't understand the second part. Leave the deprecated ones? It's just translating deprecated appearances to not deprecated ones?

Reply:
Right, `push`! Ok, so this makes sense -- it converts the replaced ones but also maintains the source.